scrapy/core/spidermw.py

```diff
@@ -60,8 +60,18 @@ def process_spider_exception(_failure):
             return _failure
 
         def process_spider_output(result):
+            def wrapper(result_iterable):
+                try:
+                    for r in result_iterable:
+                        yield r
+                except Exception as ex:
+                    exception_result = process_spider_exception(Failure(ex))
```

Comment:
I believe that

> This method is called when when a spider or process_spider_input() method (from other spider middleware) raises an exception

means that exceptions from previous `process_spider_output` methods should be handled by `process_spider_exception`; the way of getting a spider's result (either items, requests or exceptions) is through the `process_spider_output` chain. Right?
I'm rewriting some tests and adding some more to make this clear.